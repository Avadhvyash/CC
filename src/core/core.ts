// RTL
export {Dir, LayoutDirection} from './rtl/dir';

// Portals
export {
  Portal,
  PortalHost,
  BasePortalHost,
  ComponentPortal,
  TemplatePortal
} from './portal/portal';
export {
  PortalHostDirective,
  TemplatePortalDirective,
  PORTAL_DIRECTIVES,
  PortalModule,
} from './portal/portal-directives';
export {DomPortalHost} from './portal/dom-portal-host';

// Overlay
export {Overlay, OVERLAY_PROVIDERS} from './overlay/overlay';
export {OverlayContainer} from './overlay/overlay-container';
export {OverlayRef} from './overlay/overlay-ref';
export {OverlayState} from './overlay/overlay-state';
export {
  ConnectedOverlayDirective,
  OverlayOrigin,
  OVERLAY_DIRECTIVES,
  OverlayModule,
} from './overlay/overlay-directives';
export {
  OverlayConnectionPosition,
  OriginConnectionPosition
} from './overlay/position/connected-position';

// Gestures
export {MdGestureConfig} from './gestures/MdGestureConfig';

// Ripple
export {MD_RIPPLE_DIRECTIVES, MdRipple} from './ripple/ripple';

// a11y
export {
  AriaLivePoliteness,
  MdLiveAnnouncer,
  LIVE_ANNOUNCER_ELEMENT_TOKEN
} from './a11y/live-announcer';

export {
  MdUniqueSelectionDispatcher,
  MdUniqueSelectionDispatcherListener
} from './coordination/unique-selection-dispatcher';
