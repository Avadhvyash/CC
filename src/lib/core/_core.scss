// Core styles that can be used to apply material design treatments to any element.
@import 'a11y/a11y';
@import 'style/elevation';
@import 'overlay/overlay';
@import 'ripple/ripple';
@import 'option/option';
@import 'option/option-theme';
@import 'selection/pseudo-checkbox/pseudo-checkbox-theme';

// Mixin that renders all of the core styles that are not theme-dependent.
@mixin mat-core() {
  // Provides external CSS classes for each elevation value. Each CSS class is formatted as
  // `mat-elevation-z$zValue` where `$zValue` corresponds to the z-space to which the element is
  // elevated.
  @for $zValue from 0 through 24 {
    .mat-elevation-z#{$zValue} {
      @include mat-elevation($zValue);
    }
  }

  @include mat-ripple();
  @include mat-option();
  @include cdk-a11y();
  @include cdk-overlay();
}

// Mixin that renders all of the core styles that depend on the theme.
@mixin mat-core-theme($theme) {
  @include mat-ripple-theme($theme);
  @include mat-option-theme($theme);
  @include mat-pseudo-checkbox-theme($theme);

  // Wrapper element that provides the theme background when the
  // user's content isn't inside of a `md-sidenav-container`.
  .mat-app-background {
    $background: map-get($theme, background);
    background-color: mat-color($background, background);
  }

  // Marker that is used to determine whether the user has added a theme to their page.
  .mat-theme-loaded-marker {
    display: none;
  }

  // Used when disabling global scrolling.
  .cdk-global-scrollblock {
    position: fixed;

    // Necessary for iOS not to expand past the viewport.
    max-width: 100vw;

    // Note: this will always add a scrollbar to whatever element it is on, which can
    // potentially result in double scrollbars. It shouldn't be an issue, because we won't
    // block scrolling on a page that doesn't have a scrollbar in the first place.
    overflow-y: scroll;
  }
}
