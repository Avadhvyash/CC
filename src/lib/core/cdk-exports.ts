/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

export {
  // Coercing
  coerceBooleanProperty,
  coerceNumberProperty,

  // Observe Content
  ObserveContentModule,
  ObserveContent,

  // Accessibility
  A11yModule,
  FocusTrap,
  FocusTrapFactory,
  FocusTrapDeprecatedDirective,
  FocusTrapDirective,
  InteractivityChecker,
  ListKeyManager,
  CanDisable,
  AriaLivePoliteness,
  LIVE_ANNOUNCER_ELEMENT_TOKEN,
  LiveAnnouncer,
  LIVE_ANNOUNCER_PROVIDER_FACTORY,
  LIVE_ANNOUNCER_PROVIDER,
  isFakeMousedownFromScreenReader,

  // Bidi
  BidiModule,
  Dir,
  Direction,
  DIR_DOCUMENT,
  Directionality,
  DIRECTIONALITY_PROVIDER_FACTORY,

  // Keyboard
  UP_ARROW,
  DOWN_ARROW,
  RIGHT_ARROW,
  LEFT_ARROW,
  PAGE_UP,
  PAGE_DOWN,
  HOME,
  END,
  ENTER,
  SPACE,
  TAB,
  ESCAPE,
  BACKSPACE,
  DELETE,

  // Platform
  PlatformModule,
  Platform,
  getSupportedInputTypes,

  // Portal
  PortalModule,
  DomPortalHost,
  Portal,
  PortalHost,
  BasePortalHost,
  ComponentPortal,
  TemplatePortal,
  TemplatePortalDirective,
  PortalHostDirective
} from '@angular/cdk';
