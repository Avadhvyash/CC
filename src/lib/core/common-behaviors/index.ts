/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

export {MdCommonModule} from './common-module';
export {CanDisable, mixinDisabled} from './disabled';
export {CanColor, mixinColor, ThemePalette} from './color';
export {CanDisableRipple, mixinDisableRipple} from './disable-ripple';
export {HasTabIndex, mixinTabIndex} from './tabindex';
