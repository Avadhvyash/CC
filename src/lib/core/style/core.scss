// Core styles that can be used to apply material design treatments to any element.
@import '../a11y/live-announcer';
@import 'elevation';
@import '../overlay/overlay';
@import 'ripple';

// Provides external CSS classes for each elevation value. Each CSS class is formatted as
// `mat-elevation-z$zValue` where `$zValue` corresponds to the z-space to which the element is
// elevated.
@for $zValue from 0 through 24 {
  .mat-elevation-z#{$zValue} {
    @include mat-elevation($zValue);
  }
}
