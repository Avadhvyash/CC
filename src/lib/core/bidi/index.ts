/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {ModuleWithProviders, NgModule} from '@angular/core';
import {Dir} from './dir';
import {Directionality, DIRECTIONALITY_PROVIDER} from './directionality';

export {
  Directionality,
  DIRECTIONALITY_PROVIDER,
  Direction
} from './directionality';
export {Dir} from './dir';

@NgModule({
  exports: [Dir],
  declarations: [Dir],
  providers: [Directionality]
})
export class BidiModule {
  /** @deprecated */
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: BidiModule,
      providers: [DIRECTIONALITY_PROVIDER]
    };
  }
}
