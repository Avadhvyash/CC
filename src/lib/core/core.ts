/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {NgModule} from '@angular/core';
import {A11yModule, BidiModule, ObserveContentModule, PortalModule} from '@angular/cdk';
import {MdLineModule} from './line/line';
import {MdOptionModule} from './option/index';
import {OverlayModule} from './overlay/index';
import {MdSelectionModule} from './selection/index';
import {MdRippleModule} from './ripple/index';

// Re-exports of the CDK to avoid breaking changes.
export * from './cdk-exports';

// Option
export * from './option/index';

// Overlay
export * from './overlay/index';

// Gestures
export {GestureConfig} from './gestures/gesture-config';
// Explicitly specify the interfaces which should be re-exported, because if everything
// is re-exported, module bundlers may run into issues with treeshaking.
export {HammerInput, HammerManager} from './gestures/gesture-annotations';

// Ripple
export * from './ripple/index';

// Selection
export * from './selection/selection';

export {
  UniqueSelectionDispatcher,
  UniqueSelectionDispatcherListener,
  UNIQUE_SELECTION_DISPATCHER_PROVIDER,
} from './coordination/unique-selection-dispatcher';

export {MdLineModule, MdLine, MdLineSetter} from './line/line';

// Style
export * from './style/index';

// Misc
export {ComponentType} from './overlay/generic-component-type';

export * from './compatibility/compatibility';

// Animation
export * from './animation/animation';

// Selection
export * from './selection/index';

// Compatibility
export {CompatibilityModule, NoConflictStyleCompatibilityMode} from './compatibility/compatibility';

// Common material module
export {MdCommonModule, MATERIAL_SANITY_CHECKS} from './common-behaviors/common-module';

// Datetime
export * from './datetime/index';

// Placeholder
export {
  FloatPlaceholderType,
  PlaceholderOptions,
  MD_PLACEHOLDER_GLOBAL_OPTIONS
} from './placeholder/placeholder-options';

// Error
export {
  ErrorStateMatcher,
  ErrorOptions,
  MD_ERROR_GLOBAL_OPTIONS,
  defaultErrorStateMatcher,
  showOnDirtyErrorStateMatcher
} from './error/error-options';

@NgModule({
  imports: [
    MdLineModule,
    BidiModule,
    MdRippleModule,
    ObserveContentModule,
    PortalModule,
    OverlayModule,
    A11yModule,
    MdOptionModule,
    MdSelectionModule,
  ],
  exports: [
    MdLineModule,
    BidiModule,
    MdRippleModule,
    ObserveContentModule,
    PortalModule,
    OverlayModule,
    A11yModule,
    MdOptionModule,
    MdSelectionModule,
  ],
})
export class MdCoreModule {}
