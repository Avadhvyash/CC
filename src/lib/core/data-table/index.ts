import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {
  CdkRowCell,
  CdkHeaderRowCell,
  CdkRow,
  CdkHeaderRowPlaceholder,
  CdkRowPlaceholder,
  CdkColumnDef,
  CdkHeaderCellDef,
  CdkCellOutlet,
  CdkRowCellDef,
  CdkRowDef,
  CdkTable,
  CdkHeaderRow,
  CdkHeaderRowDef
} from './data-table';

export * from './data-source';
export * from './data-table';

@NgModule({
  imports: [CommonModule],
  exports: [
    CdkTable, CdkRowDef, CdkRowCellDef, CdkCellOutlet, CdkHeaderCellDef,
    CdkColumnDef, CdkRowCell, CdkRow,
    CdkHeaderRowCell, CdkHeaderRow, CdkHeaderRowDef],
  declarations: [
    CdkTable, CdkRowDef, CdkRowCellDef, CdkCellOutlet, CdkHeaderCellDef,
    CdkColumnDef, CdkRowCell, CdkRow,
    CdkHeaderRowCell, CdkHeaderRow, CdkHeaderRowDef,
    CdkRowPlaceholder, CdkHeaderRowPlaceholder,
  ]

})
export class CdkDataTableModule { }
