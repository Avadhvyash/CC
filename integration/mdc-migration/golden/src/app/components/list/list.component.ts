import {Component} from '@angular/core';

@Component({
  selector: 'list-example',
  templateUrl: './list.component.html',
  styleUrls: ['./list.component.scss'],
})
export class ListComponent {}
