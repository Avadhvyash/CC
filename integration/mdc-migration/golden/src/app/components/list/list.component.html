<h2>List example</h2>
<mat-list role="list">
  <mat-list-item role="listitem">
    <mat-icon matListItemIcon>account_circle</mat-icon>
    <div matListItemTitle>Account Information</div>
  </mat-list-item>
  <mat-list-item role="listitem">
    <mat-icon matListItemIcon>settings</mat-icon>
    <div matListItemTitle>Settings</div>
  </mat-list-item>
  <mat-list-item role="listitem">
    <mat-icon matListItemIcon>logout</mat-icon>
    <div matListItemTitle>Log Out</div>
  </mat-list-item>
</mat-list>
